state_test.go

```diff
@@ -4781,3 +4783,43 @@ func TestDBSchemaUpdate(t *testing.T) {
 	assert.Equal(t, datastore.StateDataVersion, sd.Version)
 	assert.False(t, sd.UpdateInfo.HasDBSchemaUpdate)
 }
+
+func TestAutomaticReboot(t *testing.T) {
+	tempDir, _ := ioutil.TempDir("", "logs")
+	defer os.RemoveAll(tempDir)
+
+	DeploymentLogger = NewDeploymentLogManager(tempDir)
+	defer func() {
+		DeploymentLogger.Disable()
+		DeploymentLogger = nil
+	}()
+
+	log.Log = log.New()
+	log.AddHook(NewDeploymentLogHook(DeploymentLogger))
+	defer func() {
+		log.Log = log.New()
```

Comment:
Why do we do this?

Reply:
Actually needing to do this is an indication that there is a different test which is not cleaning up after itself (in particular the first `log.New()`). The cleanup is needed because `log` doesn't have a way to unregister the hook afterwards, so we just have to create a new `log.Log` instead.

I can add a comment about it.

Reply:
I see. Thanks for the clarification. A comment would be nice, so that may encourage someone to do the proper clean-up in the future.